src/particles/wakefields/WakePush.cpp

```diff
@@ -0,0 +1,81 @@
+#include "WakePush.H"
+
+#include <ablastr/particles/NodalFieldGather.H>
+#include <AMReX_BLProfiler.H>
+#include <AMReX_REAL.H>
+#include <AMReX_SPACE.H>
+
+namespace impactx::wakepush
+{
+    void WakePush(ImpactXParticleContainer & pc, const std::vector<double>& convoluted_wakefield, amrex::ParticleReal const slice_ds)
+    {
+        BL_PROFILE("impactx::wakepush::WakePush");
+
+        using namespace amrex::literals;
+
+        //Loop over refinement levels
+        int const nLevel = pc.finestLevel();
+        for (int lev = 0; lev <= nLevel; ++lev)
+        {
+            //Loop over all particle boxes
+            using ParIt = ImpactXParticleContainer::iterator;
+
+#ifdef AMREX_USE_OMP
+#pragma omp parallel if (amrex::Gpu::notInLaunchRegion())
+#endif
+
+            for (ParIt pti(pc, lev); pti.isValid(); ++pti)
+            {
+                const int np = pti.numParticles();
+
+                //Physical constants and reference quantities
+                amrex::ParticleReal const c0_SI = 2.99792458e8;
```

Comment:
```suggestion
```

This can be removed by using the internally-stored value of c in the ABLASTR library.